guardian/atoms-rendering
# Enabling privacy-enhanced mode for youtube videos rendered in atoms

## What does this change?

The embedded youtube player supports a 'privacy-enhanced mode' :

- Privacy-enhanced mode allows you to embed YouTube videos without using cookies that track viewing behaviour. This means that no activity is collected to personalise the viewing experience. Instead, video recommendations are contextual and related to the current video. Videos playing in privacy-enhanced mode won't influence the viewer's browsing experience on YouTube.

We want to enable this feature as part of on going work to improve our users privacy when they use our products.

Enhanced privacy mode is enabled by using the domain www.youtube-nocookie.com in the src url for iframe used to embed the youtube video.

For more details see:

https://support.google.com/youtube/answer/171780?hl=en-GB

This change will enable privacy-enhanced mode for youtube videos rendered by the YoutubeAtom component.

## How to test

This change can be tested once the upgraded library has been added to one of the projects that use it, eg dot-com rendering.

Visit a page which has a youtube video embedded specifically as an atom, eg:

https://www.theguardian.com/politics/live/2020/nov/26/uk-coronavirus-live-average-pay-covid-england-tiers-boris-johnson-scotland

Open the developer tools for the page and find the iframe for the embedded youtube video at the top of the page.

The src url of the iframe should have the domain www.youtube-nocookie.com.

<img width="1035" alt="Screenshot 2020-12-08 at 17 00 17" src="https://user-images.githubusercontent.com/289928/101516040-e96b9c80-3976-11eb-9d05-45bbce4d5789.png">
